precompiles/assets/query.go

```
@@ -0,0 +1,116 @@
+	tokenInfo, err := p.assetsKeeper.GetStakingAssetInfo(ctx, assetID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Pack the values into the struct
+	result := TokenInfo{
+		Name:          tokenInfo.AssetBasicInfo.Name,
+		Symbol:        tokenInfo.AssetBasicInfo.Symbol,
+		ClientChainID: clientChainID,
+		TokenID:       tokenID,
+		Decimals:      uint8(tokenInfo.AssetBasicInfo.Decimals),
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion uint32 -> uint8

(gosec)